### PR TITLE
[AMORO-2083]exclude org.apache.zookeeper in ams/server/pom.xml introdeced by other depedency except for ams-api

### PR DESCRIPTION
**Pull request by xieyi888**
## Why are the changes needed?

Close https://github.com/NetEase/amoro/issues/2083.

## Brief change log
curator-framework depended on zookeeper-3.7.1 in ams-api
zookeeper-3.4.14 is introduced by spark-core_2.12, it should be excluded.

see https://github.com/NetEase/amoro/issues/2083

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / no)
no
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by wangtaohz** [comment]
Thanks for your report! :+1:
It is a feasible solution to exclude zookeeper from spark-core_2.12, but it is not stable enough and there may be other dependencies introducing zookeeper. 

Can you verify that the zookeeper is introduced by ams-api after this excluding? Such as using `mvn dependency:tree`.

**Comment by baiyangtx** [comment]
> Thanks for your report! 👍 It is a feasible solution to exclude zookeeper from spark-core_2.12, but it is not stable enough and there may be other dependencies introducing zookeeper.
> 
> Can you verify that the zookeeper is introduced by ams-api after this excluding? Such as using `mvn dependency:tree`. @xieyi888

We have shaded the dependencies related to Zookeeper and Curator when packaging in the ams/api module. Should our other modules directly use the shaded packages we created?

**Comment by xieyi888** [comment]
> Thanks for your report! 👍 It is a feasible solution to exclude zookeeper from spark-core_2.12, but it is not stable enough and there may be other dependencies introducing zookeeper.
> 
> Can you verify that the zookeeper is introduced by ams-api after this excluding? Such as using `mvn dependency:tree`. @xieyi888
Thanks,  I had exclude zookeeper in other dependency 

before change
mvn dependency:tree
```
[INFO] -----------------< com.netease.amoro:amoro-ams-server >-----------------
[INFO] Building Amoro Project AMS Server 0.6.0-SNAPSHOT                 [34/35]
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-dependency-plugin:3.3.0:tree (default-cli) @ amoro-ams-server ---
[INFO] com.netease.amoro:amoro-ams-server:jar:0.6.0-SNAPSHOT
[INFO] +- com.netease.amoro:amoro-core:jar:0.6.0-SNAPSHOT:compile
[INFO] |  +- com.netease.amoro:amoro-ams-api:jar:0.6.0-SNAPSHOT:compile (version managed from 0.6.0-SNAPSHOT)
[INFO] |  |  \- (org.apache.zookeeper:zookeeper:jar:3.7.1:compile - omitted for conflict with 3.6.2)
[INFO] |  \- org.apache.hadoop:hadoop-auth:jar:3.2.2:compile (version managed from 3.2.2)
[INFO] |     \- (org.apache.zookeeper:zookeeper:jar:3.4.13:compile - omitted for conflict with 3.6.2)
[INFO] +- com.netease.amoro:amoro-hive:jar:0.6.0-SNAPSHOT:compile
[INFO] |  \- org.apache.hive:hive-metastore:jar:3.1.1:compile (version managed from 3.1.1)
[INFO] |     +- org.apache.hbase:hbase-client:jar:2.0.0-alpha4:compile
[INFO] |     |  \- (org.apache.zookeeper:zookeeper:jar:3.4.10:compile - omitted for conflict with 3.6.2)
[INFO] |     +- co.cask.tephra:tephra-core:jar:0.6.0:compile
[INFO] |     |  \- org.apache.twill:twill-zookeeper:jar:0.6.0-incubating:compile
[INFO] |     |     \- (org.apache.zookeeper:zookeeper:jar:3.4.5:compile - omitted for conflict with 3.6.2)
[INFO] |     \- co.cask.tephra:tephra-hbase-compat-1.0:jar:0.6.0:compile
[INFO] |        \- org.apache.hbase:hbase-server:jar:1.0.1.1:provided
[INFO] |           \- (org.apache.zookeeper:zookeeper:jar:3.4.6:provided - omitted for conflict with 3.6.2)
[INFO] +- org.apache.spark:spark-core_2.12:jar:3.3.2:compile
[INFO] |  \- org.apache.zookeeper:zookeeper:jar:3.6.2:compile
[INFO] +- com.netease.amoro:amoro-ams-api:jar:tests:0.6.0-SNAPSHOT:test
[INFO] |  \- (org.apache.zookeeper:zookeeper:jar:3.7.1:test - omitted for conflict with 3.6.2)
[INFO] \- org.apache.curator:curator-test:jar:5.2.1:test
[INFO]    \- (org.apache.zookeeper:zookeeper:jar:3.6.3:test - omitted for conflict with 3.6.2)
```
after change
mvn dependency:tree
```
[INFO] -----------------< com.netease.amoro:amoro-ams-server >-----------------
[INFO] Building Amoro Project AMS Server 0.6.0-SNAPSHOT                 [34/35]
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-dependency-plugin:3.3.0:tree (default-cli) @ amoro-ams-server ---
[INFO] com.netease.amoro:amoro-ams-server:jar:0.6.0-SNAPSHOT
[INFO] +- com.netease.amoro:amoro-core:jar:0.6.0-SNAPSHOT:compile
[INFO] |  \- com.netease.amoro:amoro-ams-api:jar:0.6.0-SNAPSHOT:compile (version managed from 0.6.0-SNAPSHOT)
[INFO] |     \- (org.apache.zookeeper:zookeeper:jar:3.7.1:compile - omitted for duplicate)
[INFO] +- com.netease.amoro:amoro-ams-api:jar:tests:0.6.0-SNAPSHOT:test
[INFO] |  \- org.apache.zookeeper:zookeeper:jar:3.7.1:compile
[INFO] \- org.apache.curator:curator-test:jar:5.2.1:test
[INFO]    \- (org.apache.zookeeper:zookeeper:jar:3.6.3:test - omitted for conflict with 3.7.1)
```

**Comment by xieyi888** [comment]
> > Thanks for your report! 👍 It is a feasible solution to exclude zookeeper from spark-core_2.12, but it is not stable enough and there may be other dependencies introducing zookeeper.
> > Can you verify that the zookeeper is introduced by ams-api after this excluding? Such as using `mvn dependency:tree`. @xieyi888
> 
> We have shaded the dependencies related to Zookeeper and Curator when packaging in the `ams/api` module. Should other modules directly use the shaded packages we created?

As I test, It seems could not use com.netease.arctic.shaded.org.apache.curator from ams-api in ams-server module, because curator/zookeeper only be shaded after package

**Comment by xieyi888** [comment]
> > Thanks for your report! 👍 It is a feasible solution to exclude zookeeper from spark-core_2.12, but it is not stable enough and there may be other dependencies introducing zookeeper.
> > Can you verify that the zookeeper is introduced by ams-api after this excluding? Such as using `mvn dependency:tree`. @xieyi888
> 
> We have shaded the dependencies related to Zookeeper and Curator when packaging in the `ams/api` module. Should other modules directly use the shaded packages we created?

As I test, It seems could not use com.netease.arctic.shaded.org.apache.curator from ams-api in ams-server module, because curator/zookeeper only be shaded after package

**Comment by zhoujinsong** [comment]
I think the main reason why `ams-server` does not have dependent on the shade dependencies is because we accessed the curator and zookeeper classes directly in the ams-server, rather than using the classes wrapped by the ams-api.

File: core/pom.xml [hunk 1]
```diff
@@ -96,6 +96,12 @@
         <dependency>
             <groupId>org.apache.hadoop</groupId>
             <artifactId>hadoop-auth</artifactId>
+            <exclusions>
```
Comment (zhoujinsong) [inline core/pom.xml:99]: Maybe we can manage this exclusion in the `DependencyManagement` block of parent pom too?

Comment (xieyi888) [reply core/pom.xml]: > Maybe we can manage this exclusion in the `DependencyManagement` block of parent pom too?

Thanks for your remind, done

> Maybe we can manage this exclusion in the `DependencyManagement` block of parent pom too?

Thanks for your remind, done

**Comment by baiyangtx** [comment]
> > > Thanks for your report! 👍 It is a feasible solution to exclude zookeeper from spark-core_2.12, but it is not stable enough and there may be other dependencies introducing zookeeper.
> > > Can you verify that the zookeeper is introduced by ams-api after this excluding? Such as using `mvn dependency:tree`. @xieyi888
> > 
> > 
> > We have shaded the dependencies related to Zookeeper and Curator when packaging in the `ams/api` module. Should other modules directly use the shaded packages we created?
> 
> As I test, It seems could not use com.netease.arctic.shaded.org.apache.curator from ams-api in ams-server module, because curator/zookeeper only be shaded after package

 We can provide a utility that depends on zk in the ams-api module, and the server module can directly depend on the methods provided by the ams-api module. This way, we can avoid the problem of the server module not being able to directly access the shaded zk framework.

**Comment by xieyi888** [comment]
> > > > Thanks for your report! 👍 It is a feasible solution to exclude zookeeper from spark-core_2.12, but it is not stable enough and there may be other dependencies introducing zookeeper.
> > > > Can you verify that the zookeeper is introduced by ams-api after this excluding? Such as using `mvn dependency:tree`. @xieyi888
> > > 
> > > 
> > > We have shaded the dependencies related to Zookeeper and Curator when packaging in the `ams/api` module. Should other modules directly use the shaded packages we created?
> > 
> > 
> > As I test, It seems could not use com.netease.arctic.shaded.org.apache.curator from ams-api in ams-server module, because curator/zookeeper only be shaded after package
> 
> We can provide a utility that depends on zk in the ams-api module, and the server module can directly depend on the methods provided by the ams-api module. This way, we can avoid the problem of the server module not being able to directly access the shaded zk framework.

Yes, It's a good solution! I had create a issue https://github.com/NetEase/amoro/issues/2133

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Pull request merged**